tmva/tmva/inc/TMVA/RTensor.hxx

```diff
@@ -228,7 +228,7 @@ public:
       fSize = Internal::GetSizeFromShape(shape);
       fStrides = Internal::ComputeStridesFromShape(shape, layout);
       fContainer = std::make_shared<Container_t>(fSize);
-      fData = &(*fContainer->begin());
+      fData = fContainer->data();
```

Comment:
Why is this a problem?
I think here Stefan used  the iterator interface to be more generic for `Container_t`. 
Maybe we can require that `Container_t` implements `data()`, but then we need to fix also in other placed (e.g. line 217)

Reply:
As explained in the commit log:
> With and empty vector, vector::begin() == vector::end() and one cannot dereference end(), [as explained in the standard](https://en.cppreference.com/w/cpp/container/vector/end)

Now, if you have another solution, fine with me.

Reply:
Yes, the same change needs to be done in line 217 and line 231.

From the documentation of RTensor, it seems to me that the alternative container that one would use is `std::span<>` for non-owning memory:
https://github.com/root-project/root/blob/master/tmva/tmva/inc/TMVA/RTensor.hxx#L154

The `std::span` also implements a `data()` function. It is very reasonable to require that all containers wrapping contiguous memory have such a function.

Reply:
@lmoneta, would Bertrands change be okay for you, if applied consistently to the file?

(by the way, if one `really` would have liked to be general, then `std::begin()` should have been used anyway instead of the `begin()` member function, to also cover C-style arrays etc.